app/code/Mykhailok/SupportChat/CustomerData/SupportChat.php

```diff
@@ -36,50 +28,46 @@ class Load extends \Magento\Framework\App\Action\Action implements
      */
     private $messageAuthor;
 
+    /**
+     * @var \Magento\Framework\App\RequestInterface
+     */
+    private $request;
+
     public function __construct(
         \Psr\Log\LoggerInterface $logger,
         \Mykhailok\SupportChat\Model\ResourceModel\ChatMessage\CollectionFactory $chatCollectionFactory,
-        \Mykhailok\SupportChat\Service\RequestValidate $requestValidate,
         \Mykhailok\SupportChat\Service\ResponseData $responseData,
-        \Magento\Framework\Message\ManagerInterface $messageManager,
         \Mykhailok\SupportChat\Model\MessageAuthor $messageAuthor,
-        Context $context
+        \Magento\Framework\App\RequestInterface $request
     ) {
-        parent::__construct($context);
         $this->logger = $logger;
         $this->chatCollectionFactory = $chatCollectionFactory;
-        $this->requestValidate = $requestValidate;
         $this->responseData = $responseData;
-        $this->messageManager = $messageManager;
         $this->messageAuthor = $messageAuthor;
+        $this->request = $request;
     }
 
     /**
      * @inheritDoc
      */
-    public function execute()
+    public function getSectionData(): array
     {
-        $result = $this->resultFactory->create(ResultFactory::TYPE_JSON);
         try {
-            $this->requestValidate->validate(false);
-
-            $limit = $this->getRequest()->getParam('limit');
+            $limit = $this->request->getParam('limit');
             $limit = (1 <= $limit) && ($limit <= 100) ? $limit : 10;
 
-            /** @var ChatCollection $chatCollection */
-            $chatCollection = $this->chatCollectionFactory->create()
+            /** @var \Mykhailok\SupportChat\Model\ResourceModel\ChatMessage\Collection $chatCollection */
+            $chatCollection = $this->chatCollectionFactory->create();
+            $chatCollection
                 ->addChatHashFilter($this->messageAuthor->getHash())
-                ->setPageSize($limit);
+                ->setPageSize($limit)
+                ->setOrder($chatCollection->getResource()->getIdFieldName(), Collection::SORT_ORDER_DESC);
             $this->responseData->prepareResponseData($chatCollection);
```

Comment:
@mykhailok-dv This code is called only here, so must be moved to this class as a private method. See a comment to the `Save` controller as well.